Core/src/org/sleuthkit/autopsy/datasourcesummary/datamodel/DataSourceInfoUtilities.java

```diff
@@ -133,9 +139,126 @@ static String getMetaFlagsContainsStatement(TSK_FS_META_FLAG_ENUM flag) {
         return "meta_flags & " + flag.getValue() + " > 0";
     }
 
+    /**
+     * Enum for specifying the sort order for getAttributes.
+     */
+    enum SortOrder {
+        DECENDING,
+        ASCENDING
+    }
+
+    /**
+     * Return a list of artifacts that have been sorted by their attribute of
+     * attributeType.
+     *
+     * Sorting on attributes of type byte[] and JSON is not currently supported.
+     *
+     * @param skCase        SleuthkitCase instance.
+     * @param artifactType  Type of artifacts to sort.
+     * @param dataSource    Data Source that the artifact belongs to.
+     * @param attributeType Attribute type to sort by.
+     * @param sortOrder     Sort order of the attributes, either ascending or
+     *                      descending.
+     * @param maxCount      Maximum number of results to return. To return all
+     *                      values maxCount should be -1.
+     *
+     * @return A list of artifacts of type artifactType sorted by the attribute
+     *         of attributeType in the given sortOrder. If no artifacts are
+     *         found an empty list will be returned.
+     *
+     * @throws TskCoreException
+     */
+    static List<BlackboardArtifact> getArtifacts(SleuthkitCase skCase, BlackboardArtifact.Type artifactType, DataSource dataSource, BlackboardAttribute.Type attributeType, SortOrder sortOrder, int maxCount) throws TskCoreException {
+        if (maxCount < 1 && maxCount != -1) {
+            throw new IllegalArgumentException("Invalid maxCount passed to getArtifacts, value must be at greater 0");
+        }
+
+        TreeMap<BlackboardAttribute, BlackboardArtifact> sortedMap = new TreeMap<>(new AttributeComparator(sortOrder));
```

Comment:
If an artifact does not have the attribute being sorted on it is not included.  Why would artifacts that doesn't have the attribute be of interest?  

You are correct, the TreeMap doesn't allow for multiple of the same key.  I will fix that.

Reply:
Ok.  I think it would be good to document that behavior in the comment.  I initially thought it was going to behave comparable to a sql sort where rows with null for the sort value are still included.  It would be good to update the comment for the 'maxCount' parameter for the same reason.